daffodil-lib/src/main/scala/org/apache/daffodil/lib/schema/annotation/props/Properties.scala

```diff
@@ -133,6 +116,24 @@ abstract class Enum[A] extends EnumBase with Converter[String, A] {
 
   def apply(name: String, context: ThrowsSDE): A
 } // end class
+object Enum {
+  trait Value[A] extends EnumValueBase { self: A =>
```

Comment:
Do we need the type parameter? To use the new `Enum.Value` a trait now needs to duplicate it's own trait name without much gain, I think?

The only thing the `Value` trait really provides now is a default implementation of the toString function, which doesn't depend on the type parameter.

I also wonder if instead of putting it in a `Enum` object, what if we just call it `EnumValue`? Similar to how `Enum`, `EnumBase`, `EnumValueBase` are their own things?

Then this just becomes:

```scala
sealed trait Foo extends EnumValue
object Foo extends Enum[Foo] {
  ...
}
```
Seems a bit cleaner and I think doesn't lose any type safety or functionality?